canvasobject.go

```diff
@@ -105,3 +109,85 @@ type Tabbable interface {
+	// will be invoked to notify this DragSource of the outcome of the drag.
+	// This is guaranteed to happen before another invocation of DragSourceBegin may be called.
+	DragSourceBegin(*PointEvent) ([]DragItem, DragItemCursor)
+
+	// DragSourceDragged is invoked when a drag and drop continues within this DragSource
+	// and has not yet entered over a potential DropTarget.
+	DragSourceDragged(*PointEvent)
+
+	// DropPending is invoked the first time an in progress drag and drop enters
+	// over a specific potential DropTarget that may accept the drop.
+	// The items that the drop target can accept are passed to this invocation,
+	// which MAY be a subset of the items returned by DragSourceBegin.
+	// Implementations may wish to use this hook to update rendering to
+	// highlight the item(s) which may be about to be dropped.
+	DropPending([]DragItem)
+
+	// DropUnpending is invoked when an in progress drag and drop which had previously
+	// entered a potential DropTarget (and having invoked DropPending()) leaves
+	// that DropTarget, but the drag still remains in progress.
+	// Implementations may wish to cancel any visual changes made in response
+	// to the corresponding DropPending invocation.
+	DropUnpending()
+
+	// DropCanceled is invoked when a drag and drop ends without being
+	// accepted by a DropTarget.
+	DropCanceled()
+
+	// DropCanceled is invoked when a drag and drop ends with the content
+	// being accepted by a DropTarget. Implementations may wish to update
+	// their visual rendering and backend state to reflect that the dragged
+	// content has been moved.
+	DropAccepted([]DragItem)
+}
+
+// DropTarget is implemented by CanvasObjects to signify that they can accept items dropped
+// from another CanvasObject implementing DragSource.
+type DropTarget interface {
+	// DropBegin is invoked when a drag and drop beginning from a DragSource
+	// enters over this DropTarget. Implementations should return the (sub)-set of items that
+	// the drop can accept, (possibly based on the content types of the DragItems),
+	// and optionally a non-nil cursor to visually represent the content which may be
+	// dropped onto this DropTarget.
+	// If a nil or zero-length DragItem slice is returned, this DropTarget is considered
+	// to not accept the drag and drop, and no further callbacks will be invoked for
+	// the in-progress drag and drop operation.
+	// If both the DragSource and this DropTarget return a cursor, the DropTarget
+	// cursor will be used to represent the drag and drop while it is dragged over this target.
+	DropBegin(*PointEvent, []DragItem) (accept []DragItem, cursor DragItemCursor)
+
+	// DropDragged is invoked when a drag and drop that had entered this DropTarget
+	// (resulting in a DropBegin invocation) continues to be dragged within this DropTarget.
+	// Implementations may wish to update rendering to show where the dragged content
+	// would land if dropped.
+	DropDragged(*PointEvent)
```

Comment:
Is this and `DropEnd` needed? we already handle dragging in a non dnd operation, so if the position matters it can still be discovered...